torch_np/_detail/_flips.py

```diff
@@ -0,0 +1,52 @@
+"""Implementations of flip-based routines and related animals. 
+"""
+
+import torch
+
+from . import _scalar_types, _util
+
+
+def flip(m_tensor, axis=None):
+    # XXX: semantic difference: np.flip returns a view, torch.flip copies
+    if axis is None:
+        axis = tuple(range(m_tensor.ndim))
+    else:
+        axis = _util.normalize_axis_tuple(axis, m_tensor.ndim)
+    return torch.flip(m_tensor, axis)
+
+
+def flipud(m_tensor):
+    return torch.flipud(m_tensor)
+
+
+def fliplr(m_tensor):
+    return torch.fliplr(m_tensor)
+
+
+def rot90(m_tensor, k=1, axes=(0, 1)):
+    axes = _util.normalize_axis_tuple(axes, m_tensor.ndim)
+    return torch.rot90(m_tensor, k, axes)
+
+
+def swapaxes(tensor, axis1, axis2):
+    return torch.swapaxes(tensor, axis1, axis2)
+
+
+# https://github.com/numpy/numpy/blob/v1.24.0/numpy/core/numeric.py#L1259
+def rollaxis(tensor, axis, start=0):
+    n = tensor.ndim
+    axis = _util.normalize_axis_index(axis, n)
+    if start < 0:
+        start += n
+    msg = "'%s' arg requires %d <= %s < %d, but %d was passed in"
+    if not (0 <= start < n + 1):
+        raise _util.AxisError(msg % ("start", -n, "start", n + 1, start))
+    if axis < start:
+        # it's been removed
+        start -= 1
+    if axis == start:
+        return tensor[...]
```

Comment:
Is this missing or what is this supposed to mean? I don't think this is a construction in PyTorch.

Reply:
In numpy indexing with ellipsis creates a view (numpy view, not pytorch view):
```
In [62]: np.shares_memory(a[...], a)
Out[62]: True
```

Same in Pytorch, really:

```
In [70]: t[...]._base is t
Out[70]: True
```

What is the relevant pytorch idiom then?

Reply:
What about simply returning `tensor`. I know it's less precise, but views are handled differently in PyTorch autograd, and this would be a somewhat unnecessary pemisation.

Otherwise, the more PyTorch-y way of creating a view of a tensor would be  `t.view(t.shape)`. But again, perhaps returning `tensor` here is good enough.

Reply:
OK, let's do that, thanks. Have added a code comment, too.